crates/evm/src/decode.rs

```diff
@@ -26,52 +49,149 @@ pub fn decode_console_logs(logs: &[Log]) -> Vec<String> {
 /// This function returns [None] if it is not a DSTest log or the result of a Hardhat
 /// `console.log`.
 pub fn decode_console_log(log: &Log) -> Option<String> {
-    // NOTE: We need to do this conversion because ethers-rs does not
-    // support passing `Log`s
-    let raw_log = RawLog { topics: log.topics.clone(), data: log.data.to_vec() };
-    let decoded = match ConsoleEvents::decode_log(&raw_log).ok()? {
-        LogsFilter(inner) => format!("{}", inner.0),
-        LogBytesFilter(inner) => format!("{}", inner.0),
-        LogNamedAddressFilter(inner) => format!("{}: {:?}", inner.key, inner.val),
-        LogNamedBytes32Filter(inner) => {
-            format!("{}: {}", inner.key, B256::new(inner.val))
-        }
-        LogNamedDecimalIntFilter(inner) => {
-            let (sign, val) = inner.val.into_sign_and_abs();
-            format!(
-                "{}: {}{}",
-                inner.key,
-                sign,
-                ethers::utils::format_units(val, inner.decimals.as_u32()).unwrap()
-            )
-        }
-        LogNamedDecimalUintFilter(inner) => {
-            format!(
-                "{}: {}",
-                inner.key,
-                ethers::utils::format_units(inner.val, inner.decimals.as_u32()).unwrap()
-            )
-        }
-        LogNamedIntFilter(inner) => format!("{}: {:?}", inner.key, inner.val),
-        LogNamedUintFilter(inner) => format!("{}: {:?}", inner.key, inner.val),
-        LogNamedBytesFilter(inner) => {
-            format!("{}: {}", inner.key, inner.val)
-        }
-        LogNamedStringFilter(inner) => format!("{}: {}", inner.key, inner.val),
-        LogNamedArray1Filter(inner) => format!("{}: {:?}", inner.key, inner.val),
-        LogNamedArray2Filter(inner) => format!("{}: {:?}", inner.key, inner.val),
-        LogNamedArray3Filter(inner) => format!("{}: {:?}", inner.key, inner.val),
+    let raw_log = AlloyLog::new_unchecked(
+        log.topics.clone().into_iter().map(|h| h.to_alloy()).collect_vec(),
+        log.data.clone().0.into(),
+    );
+    if let Ok(inner) =
+        log::decode_log(raw_log.topics().iter().map(|t| WordToken::from(*t)), &raw_log.data, true)
+    {
+        return Some(inner._0)
+    } else if let Ok(inner) =
+        logs::decode_log(raw_log.topics().iter().map(|t| WordToken::from(*t)), &raw_log.data, true)
+    {
+        return Some(format!("{}", Bytes::from(inner._0)))
+    } else if let Ok(inner) = log_address::decode_log(
```

Comment:
This seems to also be decoding `uint256`s if they fit—this shouldn't be possible? e.g see this test failure on CI run: https://github.com/foundry-rs/foundry/actions/runs/6612838157/job/17960206523?pr=5986#step:9:197

Reply:
nvm—this functionality is not yet in Alloy, so reverting to abigen for now.